components/MediList/index.js

```diff
@@ -2,52 +2,93 @@ import styled from 'styled-components';
 import { useState } from 'react';
 import { dummyMedications } from '../../dummydata';
 
-
 export default function MediList() {
-  const [medikament, setMedikament] = useState('');
-  const [dosage, setDosage] = useState('');
-
-  const handleAddMedikament = (e) => {
-    e.preventDefault();
-    const newMedikament = {
-      id: dummyMedications.length + 1,
-      medicationName: medikament,
-      dosage: dosage,
-    };
-    setMedikament('');
-    setDosage('');
+  const [medications, setMedications] = useState(dummyMedications);
+  const [medication, setMedication] = useState({ medicationName: '', dosage: '' });
+  const [searchQuery, setSearchQuery] = useState('');
+
+  const handleAddMedication = (event) => {
+    event.preventDefault();
+    setMedications([...medications, { ...medication, id: Date.now() }]);
+    setMedication({ medicationName: '', dosage: '' });
+  };
+
+  const handleSearch = () => {
```

Comment:
In this and the following functions you could go with the "normal" declaration of a function instead of the arrow function